source/MaterialXGenShader/ShaderGraph.cpp

```diff
@@ -481,6 +486,109 @@ ShaderGraphPtr ShaderGraph::create(const ShaderGraph* parent, const string& name
         // Start traversal from this output
         root = output;
     }
+    else if (element->isA<Node>())
+    {
+        NodePtr node = element->asA<Node>();
+
+        NodeDefPtr nodeDef = node->getNodeDef();
+        if (!nodeDef)
+        {
+            throw ExceptionShaderGenError("Could not find a nodedef for node '" + node->getName() + "'");
+        }
+
+        graph = std::make_shared<ShaderGraph>(parent, name, element->getDocument(), context.getReservedWords());
+
+        // Create input sockets
+        graph->addInputSockets(*nodeDef, context);
+
+        // Create output sockets
+        graph->addOutputSockets(*nodeDef);
+
+        // Create this shader node in the graph.
+        const string& newNodeName = node->getName();
```

Comment:
Just being picky, but can this be wrapped up into 1 method as there are 3 steps required?

Reply:
Good point, I will change this.